src/odemis/gui/cont/streams.py

```diff
@@ -2383,19 +2385,18 @@ def _createAddStreamActions(self):
             self.add_action("EBIC", self.addEBIC)
         if main_data.cld:
             self.add_action("CL intensity", self.addCLIntensity)
+
+        # TODO: support every component in .ccds
         if main_data.ccd and main_data.lens and model.hasVA(main_data.lens, "polePosition"):
             # Some simple SPARC have a CCD which can only do rough chamber view,
             # but no actual AR acquisition. This is indicate by not having any
             # polePosition VA on the optical path.
             self.add_action("Angle-resolved", self.addAR)
 
         # On the SPARCv2, there is potentially 4 different ways to acquire a
-        # spectrum: two spectrographs, with each two ports. In practice, there
-        # are never more than 2 at the same time.
-        sptms = [main_data.spectrometer, main_data.spectrometer_int]
-        sptms = [s for s in sptms if s is not None]
-        for sptm in sptms:
-            if len(sptms) == 1:
+        # spectrum: two spectrographs, with each two ports.
```

Comment:
keep it generic regarding number of spectrographs? Maybe the next system has 3 :P
word order: `each with two ports`